src/modules/ztsi/src/so/ZtsiModule.cpp

```diff
@@ -231,105 +162,11 @@ int MmiSet(
-    if (document.Parse(payload).HasParseError())
-    {
-        OsConfigLogError(ZtsiLog::Get(), "MmiSet unabled to parse JSON payload");
-        status = EINVAL;
-        return status;
-    }
-
-    session = reinterpret_cast<Ztsi*>(clientSession);
-
-    if (0 == g_componentName.compare(componentName))
-    {
-        if (0 == g_desiredEnabled.compare(objectName))
-        {
-            if (document.IsBool())
-            {
-                status = session->SetEnabled(document.GetBool());
-            }
-            else
-            {
-                OsConfigLogError(ZtsiLog::Get(), "MmiSet %s is not of type boolean", g_desiredEnabled.c_str());
-                status = EINVAL;
-            }
-        }
-        else if (0 == g_desiredServiceUrl.compare(objectName))
-        {
-            if (document.IsString())
-            {
-                status = session->SetServiceUrl(document.GetString());
-            }
-            else
-            {
-                OsConfigLogError(ZtsiLog::Get(), "MmiSet %s is not of type string", g_desiredServiceUrl.c_str());
-                status = EINVAL;
-            }
-        }
-        else
-        {
-            OsConfigLogError(ZtsiLog::Get(), "MmiSet called with invalid objectName: %s", objectName);
-            status = EINVAL;
-        }
     }
     else
     {
-        OsConfigLogError(ZtsiLog::Get(), "MmiSet called with invalid componentName: %s", componentName);
-        status = EINVAL;
-    }
-
-    return status;
-}
-
-int SerializeJsonObject(MMI_JSON_STRING* payload, int* payloadSizeBytes, unsigned int maxPayloadSizeBytes, rapidjson::Document& document)
-{
-    int status = MMI_OK;
-
-    rapidjson::StringBuffer buffer;
-    rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
-    document.Accept(writer);
-
-    if (buffer.GetSize() > maxPayloadSizeBytes)
-    {
-        OsConfigLogError(ZtsiLog::Get(), "SerializeJsonObject failed to serialize JSON object to buffer");
-        status = E2BIG;
-        return status;
-    }
-
-    try
-    {
-        *payload = new (std::nothrow) char[buffer.GetSize()];
-        if (nullptr == *payload)
-        {
-            OsConfigLogError(ZtsiLog::Get(), "SerializeJsonPayload unable to allocate memory for payload");
-            status = ENOMEM;
-        }
-        else
-        {
-            std::fill(*payload, *payload + buffer.GetSize(), 0);
-            std::memcpy(*payload, buffer.GetString(), buffer.GetSize());
-            *payloadSizeBytes = buffer.GetSize();
-        }
-    }
-    catch (const std::exception& e)
-    {
-        OsConfigLogError(ZtsiLog::Get(), "Could not allocate payload: %s", e.what());
-        status = EINTR;
-
-        if (nullptr != *payload)
-        {
-            delete[] * payload;
-            *payload = nullptr;
-        }
-
-        if (nullptr != payloadSizeBytes)
-        {
-            *payloadSizeBytes = 0;
-        }
+        session = reinterpret_cast<Ztsi*>(clientSession);
```

Comment:
>session = reinterpret_cast<Ztsi*>(clientSession); [](http://example.com/codeflow?start=8&length=49)

What if some client calls MpiSet with a non null but invalid handle? This cast would prodice an invalid module object. Add a "IsValid()" member to the module  object to validate itself? (by size, contents, etc)

Reply:
Same for MpiGet